src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
         }
 
-        public void RegisterPlatoonBirth(PlatoonBehaviour platoon)
+        private void Start()
         {
-            _selectionManager.RegisterPlatoonBirth(platoon);
+            this._firePosReticle = (Texture2D)Resources.Load("FirePosTestTexture");
+            if (this._firePosReticle is null)
+                throw new MissingManifestResourceException("Could not locate Texture2D named 'FirePosTestTexture'.");
+
+            this._primedReticle = (Texture2D)Resources.Load("PrimedCursor");
+            if (this._primedReticle is null)
+                throw new MissingManifestResourceException("Could not locate Texture2D named `PrimedCursor`.");
+
+            this._rightClickManager = new ClickManager(
+                1,
+                this.MoveGhostsToMouse,
+                this.OnOrderHold,
+                this.ExitPurchasingMode,
+                this.MoveGhostsToMouse);
         }
 
-        public void RegisterPlatoonDeath(PlatoonBehaviour platoon)
+        private void Update()
         {
-            _selectionManager.RegisterPlatoonDeath(platoon);
+            this._selectionManager.Update(this.MouseState);
+
+            switch (this.MouseState)
+            {
+                case Modes.Purchasing:
+                    if (Util.GetTerrainClickLocation(out RaycastHit hit)
+                        && hit.transform.gameObject.name.Equals("Terrain"))
+                        this.ShowGhostUnitsAndMaybePurchase(hit);
+                    this.MaybeExitPurchasingMode();
+                    break;
+
+                case Modes.Normal:
+                    this._rightClickManager.Update();
+                    break;
+
+                case Modes.FirePosition:
+                    if (this.IsMouse(Events.Down))
+                        this._selectionManager.DispatchFirePosCommand();
+                    if (this.IsMouse(Events.Down) && !KeyDown(KeyCode.LeftShift) || this.IsMouse(Events.Down, 1))
+                        this.SetCursor(Modes.Normal);
+                    break;
+
+                case Modes.ReverseMove:
+                    if (!this.IsMouse(Events.Down))
+                        break;
+
+                    this.MoveGhostsToMouse();
+                    this._selectionManager.DispatchMoveCommand(false, MoveWaypoint.MoveMode.Reverse);
+
+                    if (this.IsMouse(Events.Down) && !KeyDown(KeyCode.LeftShift) || this.IsMouse(Events.Down))
+                        this.SetCursor(Modes.Normal);
+                    break;
+
+                case Modes.FastMove:
+                    if (!this.IsMouse(Events.Down))
+                        break;
+
+                    this.MoveGhostsToMouse();
+                    this._selectionManager.DispatchMoveCommand(false, MoveWaypoint.MoveMode.Fast);
+
+                    if (this.IsMouse(Events.Down) && !KeyDown(KeyCode.LeftShift) || this.IsMouse(Events.Down, 1))
+                        this.SetCursor(Modes.Normal);
```

Comment:
I was originally of the opinion that `this` can be used as often as one wants but this is excessive. If I had to make a choice I would prefer to ban `this` completely.

Reply:
i dont really see much advantage for using it,since the code shouldnt contain two variables with the same name and therefore using "this" should never be necessary. I would like to know why you used it this much, since it seems to just increase the length of the code and decrease the readability.

Reply:
There are certain areas of code I would look at (no, I can't think of anything off the top of my head) and be genuinely confused as to what the hell was what. The reason `this` is used so much is because of me sticking to a convention and using a tool to automatically do so. If a single word obstructs your vision of the code enough to mess with readability, I think you might need glasses. haha.